src/services/product.services.new.integration.test.ts

```diff
@@ -0,0 +1,99 @@
+// ! TODO: Rename all files. Replace "." with "-".
+import { db } from '../../prisma/db';
+import { createProduct, createUser } from '../test/test-utils';
+import { UpdateProductRequest } from '../types/product';
+import { ProductService } from './product.services';
+
+describe('Products Service', () => {
+  let userId: number;
+
+  beforeAll(async () => {
+    userId = (await createUser()).id;
+  });
+
+  afterAll(async () => {
+    await db.user.delete({
+      where: { id: userId },
+    });
```

Comment:
I don’t understand this comment. To test all the product service methods (CRUD), I need at least one user, because only a user can execute these actions. That’s why at the beginning (beforeAll), I’m creating a user, and then I execute all the integration tests related to products using this user. At the end (afterAll), I delete this user.

Reply:
nevermind :joy: